### PR TITLE
about group management

### PR DESCRIPTION
**Pull request by luotuo29**
Add some new functions for group management, modify queryGroups and newGroup for more rubust.

### CODE REVIEW

**Review by UDLD — APPROVED** [review]

File: core/src/main/java/com/cryptape/cita/protocol/system/CITAjSystemContract.java [hunk 1]
```diff
@@ -604,45 +678,33 @@ public QueryResourceResult queryResource(String fromAddr, String permissionAddr)
...
-            addresses.add(new Address(acc));
-        }
-
-        // groupName string convert to bytes32
+    public String newGroup(String groupName, List<String> accounts, String adminPrivatekey, int version, BigInteger chainId) throws Exception {
         String nameHex = Util.addUpTo64Hex(ConvertStrByte.stringToHexString(groupName));
         byte[] nameBytes = ConvertStrByte.hexStringToBytes(Numeric.cleanHexPrefix(nameHex));
-
-        // build input parameters
+        List<Address> addrsToAdd = new ArrayList<>();
```
Comment (duanyytop) [inline core/src/main/java/com/cryptape/cita/protocol/system/CITAjSystemContract.java:692]: I think `addrsToAdd` is uncomfortable, not simply. And I think variable names should tell other developers what it is not how to do.

Comment (luotuo29) [reply core/src/main/java/com/cryptape/cita/protocol/system/CITAjSystemContract.java:692]: So, what's your opinion? I found 'addrsToAdd' to beused by many places.

File: core/src/main/java/com/cryptape/cita/protocol/system/CITAjSystemContract.java [hunk 2]
```diff
@@ -604,45 +678,33 @@ public QueryResourceResult queryResource(String fromAddr, String permissionAddr)
...
-                           int version,
-                           BigInteger chainId) throws IOException {
-        // account addresses convert to Address object list
-        List<Address> addresses = new ArrayList<>(accounts.size());
-        for(String acc : accounts){
-            addresses.add(new Address(acc));
-        }
-
-        // groupName string convert to bytes32
+    public String newGroup(String groupName, List<String> accounts, String adminPrivatekey, int version, BigInteger chainId) throws Exception {
```
Comment (rainchen) [inline core/src/main/java/com/cryptape/cita/protocol/system/CITAjSystemContract.java]: `adminPrivatekey` should be `adminPrivateKey`, keep the same naming style as previous

File: core/src/main/java/com/cryptape/cita/protocol/system/CITAjSystemContract.java (hunk 1 above)
Comment (rainchen) [reply core/src/main/java/com/cryptape/cita/protocol/system/CITAjSystemContract.java:692]: `addrsToAdd` is OK for now, we should keep the same naming style. We can refactor them together later when we have a better naming style.

Comment (duanyytop) [reply core/src/main/java/com/cryptape/cita/protocol/system/CITAjSystemContract.java:692]: OK

**Review by duanyytop — APPROVED** [review]

**Comment by luotuo29** [comment]
This pr is for #125

**Review by rainchen — CHANGES_REQUESTED** [review]
please update and rebase, and fix the commit message by removing the prefix "1. "

File: core/src/main/java/com/cryptape/cita/protocol/system/CITAjSystemContract.java [hunk 3]
```diff
@@ -116,72 +116,72 @@ public long getQuotaPrice(String from) throws IOException {
     }
 
     public boolean approveNode(
-            String nodeAddr, String adminPrivatekey, int version, BigInteger chainId)
+            String nodeAddr, String adminPrivateKet, int version, BigInteger chainId)
```
Comment (rainchen) [inline core/src/main/java/com/cryptape/cita/protocol/system/CITAjSystemContract.java]: `adminPrivateKet` is a typo, should be `adminPrivateKey`

Comment (luotuo29) [reply core/src/main/java/com/cryptape/cita/protocol/system/CITAjSystemContract.java]: I changed

**Review by rainchen — APPROVED** [review]

**Pull request merged**